pkg/kubelet/dockershim/docker_sandbox.go

```diff
@@ -371,27 +377,31 @@ func (ds *dockerService) getIPs(podSandboxID string, sandbox *dockertypes.Contai
 	}
 
 	ips, err := ds.getIPsFromPlugin(sandbox)
-	if err == nil {
+	if err == nil && len(ips) > 0 {
 		return ips
 	}
 
+	if err != nil {
+		// TODO: trusting the docker ip is not a great idea. However docker uses
+		// eth0 by default and so does CNI, so if we find a docker IP here, we
+		// conclude that the plugin must have failed setup, or forgotten its ip.
+		// This is not a sensible assumption for plugins across the board, but if
+		// a plugin doesn't want this behavior, it can throw an error.
+		// If error happens, warn but don't return an error, as pod status
+		// should generally not return anything except fatal errors
+		// FIXME: handle network errors by restarting the pod somehow?
+		klog.Errorf("failed to read pod IP from plugin/docker: %v", err)
```

Comment:
Can you share more about why this changed from `klog.Warningf` to `klog.Errorf`?

Reply:
If error exist here, it seems real failure happens, not no-op plugin's expected result